dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/ldap/authentication/DhisBindAuthenticator.java

```diff
@@ -56,19 +58,34 @@ public DhisBindAuthenticator(BaseLdapPathContextSource contextSource, UserStore
   @Override
   public DirContextOperations authenticate(Authentication authentication) {
     User user = userStore.getUserByUsername(authentication.getName());
-
     if (user == null) {
       throw new BadCredentialsException("Incorrect user credentials");
     }
 
-    if (user.hasLdapId()) {
+    if (user.hasLdapId() && user.isExternalAuth()) {
       log.debug(
-          "Attemping username/password LDAP authentication for user: '{}'", user.getUsername());
+          "LDAP authentication attempt with username: '{}' and LDAP ID: '{}'",
+          user.getUsername(),
+          user.getLdapId());
       authentication =
           new UsernamePasswordAuthenticationToken(
               user.getLdapId(), authentication.getCredentials());
+    } else {
+      String msg =
+          format(
+              "Could not bind user to LDAP host due to missing LDAP ID or external auth: '%s'",
+              user.getUsername());
+      throw new BadCredentialsException(msg);
     }
 
     return super.authenticate(authentication);
   }
+
+  @Override
+  public void handleBindException(String userDn, String username, Throwable cause) {
+    String msg =
+        format("Failed to bind to LDAP host with DN: '%s' and username: '%s'", userDn, username);
+    log.warn(msg, cause);
+    log.debug("LDAP user bind failed", cause);
```

Comment:
is the debug necessary? it looks like it just repeats what the warn already does.